apps/moodytunes/views.py

```diff
@@ -88,3 +92,57 @@ def post(self, request, *args, **kwargs):
                 extra={'fingerprint': 'invalid_suggested_song'}
             )
             return render(request, self.template_name, context={'form': form})
+
+
+class SpotifyAuthenticationView(TemplateView):
+    template_name = 'spotify_auth.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(SpotifyAuthenticationView, self).get_context_data(**kwargs)
+
+        spotify_client = SpotifyClient()
+        context['spotify_auth_url'] = spotify_client.build_spotify_oauth_confirm_link(
+            state='user:{}'.format(self.request.user.id),
+            scopes=['playlist-modify-private']
+        )
+
+        return context
+
+
+class SpotifyAuthenticationCallbackView(View):
+
+    def get(self, request, *args, **kwargs):
+        if 'code' in request.GET:
+            code = request.GET['code']
+            state = request.GET['state']
+
+            # Parse MoodyUser primary key from state
+            user_id = state.split(':')[1]
```

Comment:
Since we are redirecting from a page on MoodyTunes, do we have a `request.user`? Let's see if the user is already assigned to the request, if so we can remove this logic and access `request.user` directly